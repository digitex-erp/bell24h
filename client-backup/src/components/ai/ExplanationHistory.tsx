import React, { useState, useEffect, useRef } from 'react';
import {
  Box,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Paper,
  Typography,
  IconButton,
  TablePagination,
  TableSortLabel,
  CircularProgress,
  useTheme,
  useMediaQuery
} from '@mui/material';
import { Refresh as RefreshIcon, Visibility as VisibilityIcon } from '@mui/icons-material';
import { format } from 'date-fns';

interface Explanation {
  id: string;
  timestamp: string;
  model: string;
  confidence: number;
  summary: string;
  features: Record<string, number>;
}

interface ExplanationHistoryProps {
  onViewExplanation: (explanation: Explanation) => void;
  refreshInterval?: number;
}

export const ExplanationHistory: React.FC<ExplanationHistoryProps> = ({
  onViewExplanation,
  refreshInterval = 30000
}) => {
  const [explanations, setExplanations] = useState<Explanation[]>([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(10);
  const [orderBy, setOrderBy] = useState<keyof Explanation>('timestamp');
  const [order, setOrder] = useState<'asc' | 'desc'>('desc');

  // Theme and responsive hooks
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
  const loadingRef = useRef<HTMLDivElement>(null);
  const tableRef = useRef<HTMLDivElement>(null);

  const fetchExplanations = async () => {
    try {
      setLoading(true);
      setError(null);
      
      const queryParams = new URLSearchParams({
        page: (page + 1).toString(),
        limit: rowsPerPage.toString(),
        sortBy: orderBy,
        sortOrder: order
      });

      const response = await fetch(`/api/explanations?${queryParams.toString()}`);
      
      if (!response.ok) {
        throw new Error('Failed to fetch explanations');
      }

      const data = await response.json();
      setExplanations(data.items);
    } catch (err) {
      setError(err instanceof Error ? err.message : 'An error occurred');
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    fetchExplanations();
    const interval = setInterval(fetchExplanations, refreshInterval);
    return () => clearInterval(interval);
  }, [page, rowsPerPage, orderBy, order, refreshInterval]);

  const handleChangePage = (event: unknown, newPage: number) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  const handleSort = (property: keyof Explanation) => {
    const isAsc = orderBy === property && order === 'asc';
    setOrder(isAsc ? 'desc' : 'asc');
    setOrderBy(property);
  };

  const handleRefresh = () => {
    fetchExplanations();
  };

  if (error) {
    return (
      <Box sx={{ p: 2 }}>
        <Typography color="error">{error}</Typography>
        <IconButton onClick={handleRefresh}>
          <RefreshIcon />
        </IconButton>
      </Box>
    );
  }

  return (
    <Box sx={{ width: '100%' }}>
      <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', mb: 2 }}>
        <Typography variant="h6">Explanation History</Typography>
        <IconButton onClick={handleRefresh} disabled={loading}>
          <RefreshIcon />
        </IconButton>
      </Box>

      <TableContainer component={Paper} ref={tableRef}>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>
                <TableSortLabel
                  active={orderBy === 'timestamp'}
                  direction={orderBy === 'timestamp' ? order : 'asc'}
                  onClick={() => handleSort('timestamp')}
                >
                  Timestamp
                </TableSortLabel>
              </TableCell>
              <TableCell>
                <TableSortLabel
                  active={orderBy === 'model'}
                  direction={orderBy === 'model' ? order : 'asc'}
                  onClick={() => handleSort('model')}
                >
                  Model
                </TableSortLabel>
              </TableCell>
              <TableCell>
                <TableSortLabel
                  active={orderBy === 'confidence'}
                  direction={orderBy === 'confidence' ? order : 'asc'}
                  onClick={() => handleSort('confidence')}
                >
                  Confidence
                </TableSortLabel>
              </TableCell>
              <TableCell>Summary</TableCell>
              <TableCell>Actions</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {loading ? (
              <TableRow>
                <TableCell colSpan={5} align="center">
                  <CircularProgress />
                </TableCell>
              </TableRow>
            ) : explanations.length === 0 ? (
              <TableRow>
                <TableCell colSpan={5} align="center">
                  No explanations available
                </TableCell>
              </TableRow>
            ) : (
              explanations.map((explanation) => (
                <TableRow key={explanation.id}>
                  <TableCell>
                    {format(new Date(explanation.timestamp), 'MMM d, yyyy HH:mm')}
                  </TableCell>
                  <TableCell>{explanation.model}</TableCell>
                  <TableCell>{(explanation.confidence * 100).toFixed(1)}%</TableCell>
                  <TableCell>{explanation.summary}</TableCell>
                  <TableCell>
                    <IconButton
                      size="small"
                      onClick={() => onViewExplanation(explanation)}
                    >
                      <VisibilityIcon />
                    </IconButton>
                  </TableCell>
                </TableRow>
              ))
            )}
          </TableBody>
        </Table>
      </TableContainer>

      <TablePagination
        rowsPerPageOptions={[5, 10, 25]}
        component="div"
        count={-1}
        rowsPerPage={rowsPerPage}
        page={page}
        onPageChange={handleChangePage}
        onRowsPerPageChange={handleChangeRowsPerPage}
      />
    </Box>
  );
};