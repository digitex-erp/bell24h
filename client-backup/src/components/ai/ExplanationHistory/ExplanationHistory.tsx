import React, { useState, useEffect, useRef } from 'react';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Paper from '@mui/material/Paper';
import CircularProgress from '@mui/material/CircularProgress';
import Alert from '@mui/material/Alert';
import TablePagination from '@mui/material/TablePagination';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import TableSortLabel from '@mui/material/TableSortLabel';
import { useTheme } from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';

// Define a basic type for an explanation item (you can expand this later)
export interface ExplanationItem {
  id: string;
  modelType: string;
  timestamp: string;
  summary: string;
}

// Define props for the component
interface ExplanationHistoryProps {
  explanations: ExplanationItem[];
  isLoading: boolean;
  error: { message: string } | null;
  // Optional props for fetching data if managed externally
  onFetchExplanations?: (params: { page: number; limit: number; sortBy: keyof ExplanationItem; sortOrder: 'asc' | 'desc' }) => Promise<{ data: ExplanationItem[]; total: number }>;
}

interface HeadCell {
  id: keyof ExplanationItem | 'actions'; // Include 'actions' or other non-data fields if needed
  label: string;
  numeric: boolean;
}

const headCells: readonly HeadCell[] = [
  { id: 'id', numeric: false, label: 'ID' },
  { id: 'modelType', numeric: false, label: 'Model Type' },
  { id: 'timestamp', numeric: false, label: 'Timestamp' },
  { id: 'summary', numeric: false, label: 'Summary' },
];

// Responsive styles for the component
const responsiveStyles = {
  tableCell: {
    padding: '16px',
    '@media (max-width: 600px)': {
      padding: '12px 8px',
      '&:first-of-type': { paddingLeft: '16px' },
      '&:last-child': { paddingRight: '16px' },
    },
  },
  tableRow: {
    '&:hover': { backgroundColor: 'action.hover' },
    '&:focus-within': {
      outline: '2px solid #4a90e2',
      outlineOffset: '-2px',
    },
    '@media (max-width: 600px)': {
      minHeight: '48px',
    },
  },
  sortLabel: {
    minHeight: '48px',
    minWidth: '48px',
    display: 'flex',
    alignItems: 'center',
    margin: '-12px -16px',
    padding: '12px 16px',
  }
};

export const ExplanationHistory: React.FC<ExplanationHistoryProps> = ({
  explanations: initialExplanations = [],
  isLoading: initialLoading = false,
  error: initialError = null,
  onFetchExplanations
}) => {
  const [explanations, setExplanations] = useState<ExplanationItem[]>(initialExplanations);
  const [loading, setLoading] = useState<boolean>(initialLoading);
  const [error, setError] = useState<string | null>(initialError ? initialError.message : null);
  
  // Theme and responsive hooks
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));

  // Pagination state
  const [page, setPage] = useState<number>(0); // 0-indexed
  const [rowsPerPage, setRowsPerPage] = useState<number>(10);
  const [totalItems, setTotalItems] = useState<number>(0);

  // Sorting state
  const [sortBy, setSortBy] = useState<keyof ExplanationItem>('timestamp'); // Default sort field
  const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('desc'); // Default sort order

  // Effect to load explanations
  useEffect(() => {
    const params = new URLSearchParams({
      page: (page + 1).toString(),
      limit: rowsPerPage.toString(),
      sortBy: sortBy as string, // API might expect string
      sortOrder: sortOrder,
    });
    console.log('[TEST_DEBUG] Effect for fetch triggered. Params:', params.toString(), 'Current sortModel:', JSON.stringify({ sortBy, sortOrder })); // DEBUG
    if (onFetchExplanations) {
      setLoading(true);
      setError(null);
      onFetchExplanations({ page, limit: rowsPerPage, sortBy, sortOrder })
        .then(response => {
          setExplanations(response.data);
          setTotalItems(response.total);
        })
        .catch(err => {
          setError(err.message || 'Failed to fetch explanations');
          setExplanations([]); // Clear data on error
          setTotalItems(0);
        })
        .finally(() => {
          setLoading(false);
        });
    } else if (initialExplanations.length > 0 && !onFetchExplanations) {
      // If onFetchExplanations is not provided, but initialExplanations are, use them.
      setExplanations(initialExplanations);
      setTotalItems(initialExplanations.length);
      setLoading(false);
    } else {
      setExplanations([]); // Ensure explanations is empty if no data source
      setTotalItems(0);
      setLoading(false);
    }
  }, [onFetchExplanations, page, rowsPerPage, sortBy, sortOrder, initialExplanations]);

  // Pagination handlers
  const handleChangePage = (event: React.MouseEvent<HTMLButtonElement> | null, newPage: number) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0); // Reset to first page
  };

  // Sorting handler
  const handleSortRequest = (property: keyof ExplanationItem) => {
    const isAsc = sortBy === property && sortOrder === 'asc';
    setSortOrder(isAsc ? 'desc' : 'asc');
    setSortBy(property);
    setPage(0); // Reset to first page on sort
  };

  const handleSortLabelClick = (property: keyof ExplanationItem) => {
    handleSortRequest(property);
  };

  const handleSortLabelKeyDown = (property: keyof ExplanationItem, event: React.KeyboardEvent<HTMLSpanElement>) => {
    if (event.key === 'Enter' || event.key === ' ') {
      event.preventDefault();
      handleSortRequest(property);
    }
  };

  // Handle keyboard navigation for the table
  const handleTableKeyDown = (e: React.KeyboardEvent) => {
    // Basic arrow key navigation could be added here if needed
    // For now, primarily for focus management and Enter/Space on rows
    if (e.key === 'ArrowDown' || e.key === 'ArrowUp') {
      // Implement row navigation if desired
    }
  };

  if (loading && explanations.length === 0) { // Show full page loader only on initial load
    return (
      <Box
        sx={{ 
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          minHeight: "200px"
        }}
        role="progressbar"
        aria-busy="true"
        aria-live="polite"
      >
        <CircularProgress />
        <Typography sx={{ marginLeft: 2 }}>Loading explanations...</Typography>
      </Box>
    );
  }

  if (error) {
    return (
      <Alert 
        severity="error" 
        sx={{ margin: 2 }}
        role="alert"
        aria-live="assertive"
      >
        {error}
      </Alert>
    );
  }
  
  return (
    <Paper 
      elevation={3} 
      sx={{ 
        padding: 2, 
        margin: 2,
        '&:focus-visible': {
          outline: '2px solid #4a90e2',
          outlineOffset: '2px',
        }
      }}
      data-testid="explanation-history-container"
      tabIndex={-1} // Makes Paper focusable if needed, but content inside should be focusable
    >
      <Typography 
        variant="h5" 
        gutterBottom 
        component="div" // More semantic for a heading
      >
        Explanation History
      </Typography>
      {explanations.length === 0 && !loading && !error ? (
        <Typography
          variant="body1"
          sx={{
            padding: 2,
            textAlign: 'center',
            color: 'text.secondary', // Use theme color for secondary text
          }}
          aria-live="assertive"
        >
          No explanations available.
        </Typography>
      ) : !error ? ( // Only render table if no error and potentially data (or loading new data)
        <TableContainer 
          tabIndex={0} // Make container focusable for keyboard nav context
          onKeyDown={handleTableKeyDown}
          sx={{
            '&:focus-visible': {
              outline: '2px solid #4a90e2', // Focus indicator for the container
              outlineOffset: '2px',
            }
          }}
        >
          <Table stickyHeader aria-label="explanation history table">
            <TableHead>
              <TableRow>
                {headCells.map((headCell) => (
                  <TableCell
                    key={headCell.id}
                    align={headCell.numeric ? 'right' : 'left'}
                    padding={headCell.id === 'actions' ? 'checkbox' : 'normal'} // Adjust padding for actions
                    sortDirection={sortBy === headCell.id ? sortOrder : false}
                    sx={responsiveStyles.tableCell}
                  >
                    <TableSortLabel
                      active={sortBy === headCell.id}
                      direction={sortBy === headCell.id ? sortOrder : 'asc'}
                      onClick={() => handleSortLabelClick(headCell.id as keyof ExplanationItem)}
                      onKeyDown={(e: React.KeyboardEvent<HTMLSpanElement>) => handleSortLabelKeyDown(headCell.id as keyof ExplanationItem, e)}
                      tabIndex={0} // Make sort label focusable
                      component="span" // Use span for proper accessibility with TableSortLabel
                      sx={responsiveStyles.sortLabel}
                      aria-label={`Sort by ${headCell.label} ${sortBy === headCell.id ? 
                        (sortOrder === 'asc' ? 'ascending' : 'descending') : ''}`}
                    >
                      {headCell.label}
                      {sortBy === headCell.id ? (
                        <Box sx={visuallyHidden}>
                          {sortOrder === 'desc' ? 'sorted descending' : 'sorted ascending'}
                        </Box>
                      ) : null}
                    </TableSortLabel>
                  </TableCell>
                ))}
              </TableRow>
            </TableHead>
            <TableBody>
              {explanations.map((exp) => (
                <TableRow 
                  hover 
                  key={exp.id}
                  tabIndex={0} // Make rows focusable
                  role="row"
                  sx={{
                    ...responsiveStyles.tableRow, // Apply responsive styles
                    '&:focus-within': { // Enhanced focus style for row
                      outline: '2px solid #4a90e2',
                      outlineOffset: '-2px',
                    },
                    '&:hover': {
                      cursor: 'pointer',
                      backgroundColor: 'action.hover', // Use theme hover color
                    },
                  }}
                  onClick={() => {
                    // Handle row click if needed
                    console.log('Row clicked:', exp.id);
                  }}
                  onKeyDown={(e: React.KeyboardEvent<HTMLTableRowElement>) => {
                    if (e.key === 'Enter' || e.key === ' ') {
                      e.preventDefault();
                      // Handle row activation if needed
                      console.log('Row activated with keyboard:', exp.id);
                    }
                  }}
                >
                  <TableCell sx={responsiveStyles.tableCell}>{exp.id}</TableCell>
                  <TableCell sx={responsiveStyles.tableCell}>{exp.modelType}</TableCell>
                  <TableCell sx={responsiveStyles.tableCell}>{new Date(exp.timestamp).toLocaleString()}</TableCell>
                  <TableCell sx={responsiveStyles.tableCell}>{exp.summary}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      ) : null}{/* Render nothing if error and no data, error is handled by Alert above */}
      { !error && explanations.length > 0 && (
        <TablePagination
          component="div"
          count={totalItems}
          page={page}
          onPageChange={handleChangePage}
          rowsPerPage={rowsPerPage}
          onRowsPerPageChange={handleChangeRowsPerPage}
          rowsPerPageOptions={isMobile ? [5, 10] : [5, 10, 25, 50]}
          labelRowsPerPage={isMobile ? '' : 'Rows:'} // Hide label on mobile for space
          labelDisplayedRows={({ from, to, count }: { from: number; to: number; count: number }) => 
            isMobile ? `${from}-${to} of ${count}` : `Showing ${from}-${to} of ${count}`
          }
          sx={{ // Responsive adjustments for pagination
            '& .MuiTablePagination-toolbar': {
              flexWrap: 'wrap', // Allow wrapping on small screens
              gap: 1, // Add gap between elements
            },
            '& .MuiTablePagination-actions': {
              margin: 0, // Remove default margin
              '& button': {
                padding: '8px', // Ensure touch target size
                minWidth: '48px', // Consistent button size
                minHeight: '48px',
              },
            },
          }}
        />
      )}
    </Paper>
  );
};

// Helper for visually hidden class (for accessibility with TableSortLabel)
const visuallyHidden = {
  border: 0,
  clip: 'rect(0 0 0 0)',
  height: 1,
  margin: -1,
  overflow: 'hidden',
  padding: 0,
  position: 'absolute' as 'absolute', // Ensure 'absolute' is treated as a literal type
  top: 20,
  width: 1,
};

export default ExplanationHistory;
