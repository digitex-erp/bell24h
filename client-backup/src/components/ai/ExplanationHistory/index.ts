export { default } from './ExplanationHistory';
export * from './ExplanationHistory';
