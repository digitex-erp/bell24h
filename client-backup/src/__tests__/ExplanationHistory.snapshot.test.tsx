import React from 'react';
import { render } from '@testing-library/react';
import ExplanationHistory from '@/components/ai/ExplanationHistory';

describe('ExplanationHistory snapshot', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<ExplanationHistory items={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
