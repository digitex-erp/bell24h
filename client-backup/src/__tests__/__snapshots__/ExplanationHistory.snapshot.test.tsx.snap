// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExplanationHistory snapshot > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="MuiBox-root css-nfgtzt"
    role="status"
  >
    <span
      aria-label="Loading explanations"
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
      title="Loading explanations"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <div
      class="MuiBox-root css-3hetpk"
    >
      Loading explanations...
    </div>
  </div>
</DocumentFragment>
`;

exports[`ExplanationHistory snapshot matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-19kzrtu"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
    >
      AI Feature Importance History
    </h6>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="explanation history table"
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              Feature
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
              scope="col"
            >
              Value
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
              scope="col"
            >
              Importance
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall css-vtdehq-MuiTableCell-root"
              colspan="3"
            >
              No explanation history available
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root css-18khl2d-MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
          id=":r1:"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
        >
          <select
            aria-label="rows per page"
            class="MuiNativeSelect-select MuiTablePagination-select MuiNativeSelect-standard MuiInputBase-input css-m3l1ig-MuiNativeSelect-select-MuiInputBase-input"
            id=":r0:"
          >
            <option
              class="MuiTablePagination-menuItem"
              value="5"
            >
              5
            </option>
            <option
              class="MuiTablePagination-menuItem"
              value="10"
            >
              10
            </option>
            <option
              class="MuiTablePagination-menuItem"
              value="25"
            >
              25
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiTablePagination-selectIcon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          0â€“0 of 0
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
